airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py

```diff
@@ -219,6 +219,13 @@ def transform_snowflake(config: Dict[str, Any]):
             dbt_config["oauth_client_id"] = credentials["client_id"]
             dbt_config["oauth_client_secret"] = credentials["client_secret"]
             dbt_config["token"] = credentials["refresh_token"]
+        elif credentials.get("private_key"):
+            f = open("private_key_path.txt", "w")
```

Comment:
nitpick: prefer `with open()... as f` over `open()... close()`; this is basically equivalent to Java's `try (OutputStream s = ...)`:
```python
with open("private_key_path.txt", "w") as f:
  f.write(credentials["private_key"])
```
(then you don't need to `f.close()` at all)